wsidicom/uid.py

```diff
@@ -70,10 +70,7 @@ def matches(self, other: "SlideUids") -> bool:
         if settings.strict_uid_check:
             return self == other
 
-        return (
-            self.study_instance == other.study_instance
-            and self.series_instance == other.series_instance
-        )
+        return self.frame_of_reference == other.frame_of_reference
```

Comment:
I think the check when having the (default) setting `strict_uid_check` is true should be that both `study` and `frame of reference` matches, and the non-strich check that only the `study` matches. The non-strict setting is usefull if the frame of reference uid is wrong.

Reply:
Okay! I pushed a fix for this. Feel free to correct me if I misunderstood anything.